tee-worker/litentry/core/credentials/src/brc20/amount_holder.rs

```diff
@@ -0,0 +1,185 @@
+	CATS,
+	BTCS,
+	Unknown,
+}
+
+pub trait BRC20AmountHolderCredential {
+	fn update_brc20_amount_holder_credential(&mut self, response_items: &Vec<ResponseItem>);
+}
+
+impl BRC20AmountHolderCredential for Credential {
+	fn update_brc20_amount_holder_credential(&mut self, response_items: &Vec<ResponseItem>) {
+		// let found_tokens: Vec<String> = items
+		// 	.iter()
+		// 	.filter_map(|item| {
+		// 		BRC20_TOKENS
+		// 			.iter()
+		// 			.find(|&&name| name == item.tick)
+		// 			.map(|name| name.to_string())
+		// 	})
+		// 	.collect();
+		// let matching_items: Vec<ResponseItem> = response_items
+		//     .iter()
+		//     .filter(|item| BRC20_TOKENS.contains(&item.tick.as_str()))
+		//     .cloned()
+		//     .collect();
+
+		for item in response_items {
+			if BRC20_TOKENS.contains(&item.tick.as_str()) {
+				let token = tick_to_brctoken(&item.tick);
+				let balance: f64 = item.overall_balance.parse().unwrap_or(0.0);
+				update_assertion(token, balance, self);
+			}
+		}
+
+		self.add_subject_info(VC_BRC20_AMOUNT_HOLDER_DESCRIPTIONS, VC_BRC20_AMOUNT_HOLDER_TYPE);
+	}
+}
+
+// TODO: the following part is exactly the same structure from 'token_balance.rs'.
+// Anyway the refactor is planned later. So continue using the same mechanism.
+fn update_assertion(token: BRCToken, balance: f64, credential: &mut Credential) {
+	let mut assertion = AssertionLogic::new_and();
+
+	let content = get_assertion_content(&token);
+	let range = get_balance_range(&token);
+	let index = BalanceRange::index(&range, balance);
+	match index {
+		Some(index) => {
+			let min = format!("{}", range[index]);
+			let max = format!("{}", range[index + 1]);
+			let min_item = AssertionLogic::new_item(content, Op::GreaterEq, &min);
+			let max_item = AssertionLogic::new_item(content, Op::LessThan, &max);
+
+			assertion = assertion.add_item(min_item);
+			assertion = assertion.add_item(max_item);
+
+			credential.credential_subject.values.push(index != 0);
+		},
+		None => {
+			let min_item = AssertionLogic::new_item(
+				content,
+				Op::GreaterEq,
+				&format!("{}", get_token_range_last(&token)),
+			);
+			assertion = assertion.add_item(min_item);
+
+			credential.credential_subject.values.push(true);
+		},
+	}
+
+	credential.credential_subject.assertions.push(assertion);
+}
+
+fn tick_to_brctoken(tick: &str) -> BRCToken {
+	match tick {
+		"ordi" => BRCToken::ORDI,
+		"sats" => BRCToken::SATS,
+		"rats" => BRCToken::RATS,
+		"MMSS" => BRCToken::MMSS,
+		"long" => BRCToken::LONG,
+		"cats" => BRCToken::CATS,
+		"BTCs" => BRCToken::BTCS,
+		_ => BRCToken::Unknown,
+	}
+}
+
+fn get_assertion_content(token: &BRCToken) -> &'static str {
+	match token {
+		BRCToken::ORDI => "$ordi_holding_amount",
+		BRCToken::SATS => "$sats_holding_amount",
+		BRCToken::RATS => "$rats_holding_amount",
+		BRCToken::MMSS => "$MMSS_holding_amount",
+		BRCToken::LONG => "$long_holding_amount",
+		BRCToken::CATS => "$cats_holding_amount",
+		BRCToken::BTCS => "$BTCs_holding_amount",
```

Comment:
Actually, can we not create new VC with $ordi_holding_amount but put token into seperate Assertion condition